rpc/http.go

```diff
@@ -132,6 +141,9 @@ func (hc *httpConn) doRequest(ctx context.Context, msg interface{}) (io.ReadClos
 	if err != nil {
 		return nil, err
 	}
+	if resp.StatusCode < 200 || resp.StatusCode >= 300 {
```

Comment:
I guess that's OK, but maybe it would be even better to check for 4xx explicitly.

Reply:
Why ?

Reply:
It doesn't really matter, but 4xx are for errors. Other codes don't necessarily indicate error. Let's forget about this, it's good to go.

Reply:
I edited my comment just as you were typing, I think

Reply:
1xx usually error, if it's surfaced to the caller, otherwise handled beneath the hood afaik
3xx redirection (either resolves to something else, or is error)
4xx client errors
5xx server errors

Reply:
Yeah, it's fine the way you did it. I was wrong.